src/web/pages/users/dialog.js

```diff
@@ -48,45 +52,73 @@ import Divider from 'web/components/layout/divider';
 import Layout from 'web/components/layout/layout';
 
 class Dialog extends React.Component {
+  dataToSave = false;
+
   constructor(...args) {
     super(...args);
 
     const {roleIds = []} = this.props;
 
     this.state = {
       confirmationDialogVisible: false,
+      confirmationDialogVisibleSA: false,
       noRoleConfirmed: false,
       roleIds,
     };
 
     this.closeConfirmationDialog = this.closeConfirmationDialog.bind(this);
+    this.closeConfirmationDialogSA = this.closeConfirmationDialogSA.bind(this);
+    this.openConfirmationDialogSA = this.openConfirmationDialogSA.bind(this);
     this.handleResumeClick = this.handleResumeClick.bind(this);
+    this.handleResumeClickSA = this.handleResumeClickSA.bind(this);
     this.handleRoleIdsChange = this.handleRoleIdsChange.bind(this);
     this.handleSaveClick = this.handleSaveClick.bind(this);
   }
 
+  openConfirmationDialog() {
+    this.setState({confirmationDialogVisible: true});
+  }
+
   closeConfirmationDialog() {
     this.setState({confirmationDialogVisible: false});
   }
 
+  openConfirmationDialogSA() {
+    this.setState({confirmationDialogVisibleSA: true});
+  }
+
+  closeConfirmationDialogSA() {
+    this.setState({confirmationDialogVisibleSA: false});
+  }
+
   handleResumeClick() {
     this.setState({noRoleConfirmed: true});
     this.closeConfirmationDialog();
   }
 
+  handleResumeClickSA(onSave) {
+    this.closeConfirmationDialogSA();
+    return onSave(this.dataToSave);
+  }
+
   handleRoleIdsChange(value) {
     this.setState({
       noRoleConfirmed: false,
       roleIds: value,
     });
   }
 
-  handleSaveClick(onSave, d) {
+  handleSaveClick(user, onSave, d) {
```

Comment:
I know this has been here before, but now would be a good time to change it: no one-letter variables, please :)